nexus/src/external_api/http_entrypoints.rs

```diff
@@ -364,6 +367,61 @@ async fn policy_update(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// Fetch the current silo's IAM policy
+#[endpoint {
+    method = GET,
+    path = "/policy",
+    tags = ["silos"],
+ }]
+pub async fn policy_view(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+) -> Result<HttpResponseOk<shared::Policy<authz::SiloRole>>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let authz_silo = opctx
+            .authn
+            .silo_required()
+            .internal_context("loading current silo")?;
+        let policy =
+            nexus.silo_fetch_policy_by_id(&opctx, authz_silo.id()).await?;
```

Comment:
Added this `by_id` fetcher because we have the ID on hand and it seems silly to fetch the silo to get the name just so we can use the existing `silo_fetch_policy`.